cmd/bnsd/x/username/username.go

```diff
@@ -49,25 +50,30 @@ func (u Username) String() string {
 }
 
 func (u Username) Validate() error {
-	if !validUsername(string(u)) {
-		return errors.Wrap(errors.ErrInput, "invalid username")
+	const usernameSeparator = `*`
+	chunks := strings.SplitN(string(u), usernameSeparator, 2)
+	if len(chunks) != 2 {
+		return errors.Wrap(errors.ErrInput, "missing separator")
 	}
 
+	var errs error
+	if !validName(chunks[0]) {
+		errs = errors.AppendField(errs, "Name", errors.ErrInput)
+	}
+	if !validNamespace(chunks[1]) {
+		errs = errors.AppendField(errs, "Namespace", errors.ErrInput)
+	}
+	return errs
+}
+
+var (
```

Comment:
pp: please define global vars and consts on top of the file. We have this as a convention in this project.

Reply:
I would prefer to not do this anymore.

Since my first days, I am always told to write const and vars on top of the file like we were programming in Python or some other language that requires a strict order of declaration. Order of declaration does not matter in Go. Thank to this, we can write the code top-down.

If I was not changing my code every time, by now majority of the code would not use this forced style. For me moving declaration to the top lowers the quality of the code. That is why I do not want to do this anymore.

I was trying to find any recommendation for this in Go documentation and I cannot find anything. What I found out in the code is that const and vars are sometimes on top and other times around the file. Before or after they are used.

What do you think?

Reply:
We had this discussion a couple of times already. My personal preferences are different, too but it is on top of the file from day 0 in this project.
Please make a proposal in an issue so that we can discuss this proper and vote. Can be something very simple.